tests/serverpod_cli_e2e_test/test/generate_with_directory_test.dart

```diff
@@ -0,0 +1,267 @@
+@Timeout(Duration(minutes: 5))
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+import 'package:uuid/uuid.dart';
+
+void main() async {
+  var tempPath = path.join(Directory.current.path, 'temp');
+  var rootPath = path.join(Directory.current.path, '..', '..');
+  var cliPath = path.join(rootPath, 'tools', 'serverpod_cli');
+
+  setUpAll(() async {
+    await Process.run(
+      'dart',
+      ['pub', 'global', 'activate', '-s', 'path', '.'],
+      workingDirectory: cliPath,
+    );
+
+    // Run command and activate again to force cache pub dependencies.
+    await Process.run(
+      'serverpod',
+      ['version'],
+      workingDirectory: cliPath,
+    );
+
+    await Process.run(
+      'dart',
+      ['pub', 'global', 'activate', '-s', 'path', '.'],
+      workingDirectory: cliPath,
+    );
```

Comment:
Why should this be necessary? Pub dependencies are cached on activate already, right?